src/telem/lib/input/mekanism/IsotopicCentrifugeInputAdapter.lua

```diff
@@ -0,0 +1,77 @@
+local o = require 'telem.lib.ObjectModel'
+local t = require 'telem.lib.util'
+
+local InputAdapter      = require 'telem.lib.InputAdapter'
+local Metric            = require 'telem.lib.Metric'
+local MetricCollection  = require 'telem.lib.MetricCollection'
+
+local IsotopicCentrifugeInputAdapter = o.class(InputAdapter)
+IsotopicCentrifugeInputAdapter.type = 'IsotopicCentrifugeInputAdapter'
+
+function IsotopicCentrifugeInputAdapter:constructor (peripheralName, categories)
+    self:super('constructor')
+
+    -- TODO this will be a configurable feature later
+    self.prefix = 'mekunicable:'
+
+    -- TODO make these constants
+    local allCategories = {
+        'basic',
+        'input',
+        'output'
+    }
+
+    if not categories then
+        self.categories = { 'basic' }
+    elseif categories == '*' then
+        self.categories = allCategories
+    else
+        self.categories = categories
+    end
+
+    -- boot components
+    self:setBoot(function ()
+        self.components = {}
+
+        self:addComponentByPeripheralID(peripheralName)
+    end)()
+end
+
+function IsotopicCentrifugeInputAdapter:read ()
+    self:boot()
+
+    local source, centrifuge = next(self.components)
+
+    local metrics = MetricCollection()
+
+    local loaded = {}
+
+    for _,v in ipairs(self.categories) do
+        -- skip, already loaded
+        if loaded[v] then
+            -- do nothing
+
+        -- Literally all we have lmao
+        elseif v == 'basic' then
```

Comment:
Recommend this category layout:

Basic:
- energy_usage
- energy_filled_percentage
- input_filled_percentage
- output_filled_percentage

Energy:
- energy
- energy_needed

Input:
- input
- input_capacity

Output:
- output
- output_capacity